MaxText/multihost_dataloading.py

```diff
@@ -108,11 +108,18 @@ def get_next_batch_sharded(local_dataset: tf.data.Dataset,
                            global_mesh: Mesh) -> jax.Array:
   """Splits the host loaded data equally over all devices."""
 
-  try:
-    local_data = local_dataset.next()
-  except tf.errors.FailedPreconditionError:
-    max_logging.log("Failed to get next data batch, retrying")
-    time.sleep(10)
+  loaded_data_success = False
+  data_load_attempts, max_data_load_attempts = 0, 30
```

Comment:
nit: putthing these constants into uppercase (and including a the `time.sleep` constant) would be better